core/editor.js

```diff
@@ -684,10 +684,13 @@
 		// 2. <td>[Cell]</td> (IE8-, Safari)
 		function isSelectedCell( range ) {
 			var start = range.startContainer,
-				end = range.endContainer;
+				end = range.endContainer,
+				startIsTr = start.is && start.is( 'tr' ),
+				startIsTdWithEqualChildCount = start.is && start.is( 'td' ) && start.equals( end ) && range.endOffset === start.getChildCount(),
```

Comment:
As you made `starIsTr` - the `startIsTd` seems to be also useful. So we will get rid of repetition and receive shorter lines of code.